src/d/actor/d_a_hitobj.cpp

```diff
@@ -5,28 +5,133 @@
 
 #include "d/actor/d_a_hitobj.h"
 #include "dolphin/types.h"
+#include "d/d_com_inf_game.h"
+#include "f_op/f_op_actor.h"
+#include "f_op/f_op_kankyo_mng.h"
+#include "f_op/f_op_actor_mng.h"
+#include "d/d_procname.h"
+
+static dCcD_SrcSph m_co_sph_src = {
+    // dCcD_SrcGObjInf
+    {
+        /* Flags             */ 0,
+        /* SrcObjAt  Type    */ 0x00000100,
+        /* SrcObjAt  Atp     */ 0,
+        /* SrcObjAt  SPrm    */ 0x0000000B,
+        /* SrcObjTg  Type    */ 0,
+        /* SrcObjTg  SPrm    */ 0x00000003,
+        /* SrcObjCo  SPrm    */ 0,
+        /* SrcGObjAt Se      */ 0,
+        /* SrcGObjAt HitMark */ 0,
+        /* SrcGObjAt Spl     */ 0,
+        /* SrcGObjAt Mtrl    */ 0,
+        /* SrcGObjAt SPrm    */ 0,
+        /* SrcGObjTg Se      */ 0,
+        /* SrcGObjTg HitMark */ 0,
+        /* SrcGObjTg Spl     */ 0,
+        /* SrcGObjTg Mtrl    */ 0,
+        /* SrcGObjTg SPrm    */ 0,
+        /* SrcGObjCo SPrm    */ 0,
+    },
+    // cM3dGSphS
+    {
+        /* Center */ 0.0f, 0.0f, 0.0f,
+        /* Radius */ 100.0f,
+    },
+};
+
+s32 hitobj_class::_create() {
+    fopAcM_SetupActor(this, hitobj_class);
+
+    int res = dComIfG_resLoad(&(this->mPhs), "Hitobj");
+    if (res == cPhs_COMPLEATE_e) {
+        this->m0298 = this->mBase.mParameters;
+
+        this->mStts.Init(0xFF, 0xFF, this);
+
+        this->mUnkownObj.Set(m_co_sph_src);
+
+        this->mUnkownObj.SetStts(&this->mStts);
+
+        this->m029a = 3;
+    }
+    return res;
+}
+
+BOOL hitobj_class::_delete() {
+    dComIfG_resDelete(&(this->mPhs), "Hitobj");
+    return TRUE;
+}
+
+BOOL hitobj_class::_isDelete() {
+    return TRUE;
+}
+
+BOOL hitobj_class::_execute() {
+    if (this->m029a != 0) {
+        this->m029a -= (short)1;
+        
+        (&(this->mUnkownObj))->SetC(this->current.pos);
+        
+        (&(&g_dComIfG_gameInfo)->play.mCcS)->Set((cCcD_Obj *)(&(this->mUnkownObj)));
+    } else {
+        fopKyM_Delete(this);
+    }
+
+    return TRUE;
+}
+
+BOOL hitobj_class::_draw() {
+    return TRUE;
+}
 
 /* 00000078-00000080       .text daHitobj_Draw__FP12hitobj_class */
-static BOOL daHitobj_Draw(hitobj_class*) {
-    /* Nonmatching */
+static BOOL daHitobj_Draw(hitobj_class* pHitobj) {
```

Comment:
```suggestion
static BOOL daHitobj_Draw(hitobj_class* i_this) {
```

the parameter containing the `this` actor for these C-style functions should usually be named `i_this`